File: src/com/dafttech/terra/graphics/lighting/Light.java
package com.dafttech.terra.graphics.lighting;

import com.dafttech.terra.game.world.Vector2;
import com.dafttech.terra.game.world.entities.Entity;
import com.dafttech.terra.graphics.AbstractScreen;

public abstract class Light {
    public abstract void drawToLightmap(AbstractScreen screen, Entity pointOfView);

    public abstract void setPosition(Vector2 p);
}

File: terra/src/main/scala/com/dafttech/terra/engine/lighting/Light.scala
package com.dafttech.terra.engine.lighting

import com.badlogic.gdx.graphics.Color
import com.dafttech.terra.engine.AbstractScreen
import com.dafttech.terra.engine.vector.Vector2d
import com.dafttech.terra.game.world.entities.Entity

abstract class Light {
  private[lighting] var color: Color = Color.WHITE

  def setColor(c: Color): Unit = {
    color = c
    color.a = 1
  }

  def getColor: Color = color

  def drawToLightmap(screen: AbstractScreen, pointOfView: Entity, pos: Vector2d)
}